src/main/java/org/immregistries/smm/transform/procedure/AlternativeBeginnings.java

```diff
@@ -8,60 +8,72 @@
 import org.immregistries.smm.transform.Transformer;
 
 public class AlternativeBeginnings extends ProcedureCommon implements ProcedureInterface {
-  
-  public static enum Field {
-    FIRST_NAME,
-    MIDDLE_NAME,
-    LAST_NAME,
-    MOTHERS_MAIDEN_NAME
+
+  public enum Field {
+                     FIRST_NAME(5, 2, false),
+                     MIDDLE_NAME(5, 3, false),
+                     LAST_NAME(5, 1, false),
+                     MOTHERS_MAIDEN_NAME(6, 1, false),
+                     MOTHERS_FIRST_NAME(6, 2, false),
+                     ADDRESS_STREET(11, 1, false),
+                     ADDRESS_CITY(11, 3, false),
+                     EMAIL(13, 4, true);
+
+    int fieldPos;
+    int subPos;
+    boolean repeatedField;
+
+    private Field(int fieldPos, int subPos, boolean repeatedField) {
+      this.fieldPos = fieldPos;
+      this.subPos = subPos;
+      this.repeatedField = repeatedField;
+    }
   }
-	
+
   private Field field;
-  
+
   public AlternativeBeginnings(Field field) {
     this.field = field;
   }
 
   public void doProcedure(TransformRequest transformRequest, LinkedList<String> tokenList)
       throws IOException {
     List<String[]> fieldsList = readMessage(transformRequest);
-    {
-      for (String[] fields : fieldsList) {
-        String segmentName = fields[0];
-        if (segmentName.equals("PID")) {
-          if (field == Field.LAST_NAME
-            || field == Field.FIRST_NAME
-            || field == Field.MIDDLE_NAME
-            || field == Field.MOTHERS_MAIDEN_NAME) {
-            
-            int fieldPos = 5;
-            int subPos = 1;
-            if (field == Field.LAST_NAME) {
-              subPos = 1;
-            } else if (field == Field.FIRST_NAME) {
-              subPos = 2;
-            } else if (field == Field.MIDDLE_NAME) {
-              subPos = 3;
-            } else if (field == Field.MOTHERS_MAIDEN_NAME) {
-              fieldPos = 6;
-              subPos = 1;
+
+    for (String[] fields : fieldsList) {
+      String segmentName = fields[0];
+      if ("PID".equals(segmentName)) {
+        if (!field.repeatedField) {
+          String value = readValue(fields, field.fieldPos, field.subPos);
+          value = varyName(value);
+          updateValue(value, fields, field.fieldPos, field.subPos);
+        } else {
+          int fieldPos = 13;
+          String[] repeatFields = readRepeats(fields, fieldPos);
+          int pos = 0;
+          for (String value : repeatFields) {
+            int subPos = 4;
+
+            String email = readRepeatValue(value, subPos);
```

Comment:
It's probably not good that it assumes that a repeating field is always an email. Maybe that will always be true, but perhaps it won't be in the future.

Reply:
Worked on this a bit in my 2476 branch. I went through the procedure classes supporting repeating fields, and most of them only support email right now, I got rid of the hardcoded positions to start with, relying on the current Field. Let me know what you think of this. Adding another repeated field to these classes would require work no matter what though.